### ISSUE TITLE
CodeGenerationException when initializing grainclient.

### ISSUE THREAD

**Issue by onionhammer** [dotnet/orleans#3217]
Framework: net461
Version: 1.5.0

Server: net461, running in Cloud Service (worker role)
Client: ASP.NET web app (net461), calls GrainClient.Initialize()

Results in 
```
Exception Details: Orleans.CodeGenerator.CodeGenerationException: (1,442611): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes'.
(1,375466): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes'.
(1,443357): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes'.
(1,13831): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes'.
(1,177096): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes'.
(1,177467): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes'.
(1,554697): error CS0433: The type 'ValueTuple<T1, T2>' exists in both 'System.ValueTuple, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' and 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
(1,352937): error CS0433: The type 'ValueTuple<T1, T2>' exists in both 'System.ValueTuple, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' and 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
(1,353786): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes'.
(1,354197): error CS0012: The type 'Enum' is defined in an assembly that is not referenced. You must add a reference to assembly 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e, Retargetable=Yes'.
..snip..
```

Orleans Packages included in client:
Microsoft.Orleans.Client
Microsoft.Orleans.Core
Microsoft.Orleans.OrleansAzureUtils
Microsoft.Orleans.OrleansCodeGenerator
Microsoft.Orleans.OrleansProviders
Microsoft.Orleans.OrleansRuntime

This is actually a very long error

![image](https://user-images.githubusercontent.com/969938/28088109-43ddda10-664a-11e7-8844-f57b086c677a.png)

**Comment by galvesribeiro** [comment]
Maybe a dumb question but, does Cloud Services have 461 pre-installed? Last time I checked it was required an Startup Script to do so... Those errors looks like missing framework and the redirected ones don't have the proper type...

**Comment by galvesribeiro** [comment]
Yeah, just confirmed... According to [this doc](https://docs.microsoft.com/en-us/azure/cloud-services/cloud-services-dotnet-install-dotnet) you need to install 4.6.1 yourself since the most updated GuestOS has 4.6. That link show how to do it in an _automated_ way.

**Comment by onionhammer** [comment]
@galvesribeiro Yes, this is running in the simulator on my machine, not in the cloud (yet).

**Comment by kogir** [comment]
If you're using GuestOS Family 5, 4.6.1 is installed by default: https://docs.microsoft.com/en-us/azure/cloud-services/cloud-services-guestos-update-matrix#family-5-releases

We're using it successfully with Orleans 1.5 in production right now.

**Comment by ReubenBond** [comment]
@onionhammer are you able to share the installed versions of all packages?

**Comment by onionhammer** [comment]
Here's the packages.config from the client erroring out

```
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="Antlr" version="3.5.0.2" targetFramework="net46" />
  <package id="Autofac" version="3.5.2" targetFramework="net46" />
  <package id="Autofac.Mvc5" version="3.3.4" targetFramework="net46" />
  <package id="AutoMapper" version="4.1.1" targetFramework="net46" />
  <package id="BuildWebCompiler" version="1.11.347" targetFramework="net46" />
  <package id="Chart.js" version="1.0.2" targetFramework="net46" />
  <package id="EntityFramework" version="6.1.3" targetFramework="net46" />
  <package id="FontAwesome" version="4.7.0" targetFramework="net46" />
  <package id="Hyak.Common" version="1.1.3" targetFramework="net46" />
  <package id="jQuery" version="2.2.4" targetFramework="net46" />
  <package id="jQuery.Validation" version="1.16.0" targetFramework="net46" />
  <package id="LinqKit.EntityFramework" version="1.1.9.0" targetFramework="net46" />
  <package id="Microsoft.ApplicationInsights" version="2.3.0" targetFramework="net46" />
  <package id="Microsoft.ApplicationInsights.Agent.Intercept" version="2.0.7" targetFramework="net46" />
  <package id="Microsoft.ApplicationInsights.DependencyCollector" version="2.3.0" targetFramework="net46" />
  <package id="Microsoft.ApplicationInsights.JavaScript" version="0.22.9-build00167" targetFramework="net46" />
  <package id="Microsoft.ApplicationInsights.PerfCounterCollector" version="2.3.0" targetFramework="net46" />
  <package id="Microsoft.ApplicationInsights.Web" version="2.3.0" targetFramework="net46" />
  <package id="Microsoft.ApplicationInsights.Web.TelemetryChannel" version="1.1.0" targetFramework="net46" />
  <package id="Microsoft.ApplicationInsights.WindowsServer" version="2.3.0" targetFramework="net46" />
  <package id="Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel" version="2.3.0" targetFramework="net46" />
  <package id="Microsoft.AspNet.Identity.Core" version="2.2.1" targetFramework="net46" />
  <package id="Microsoft.AspNet.Identity.Owin" version="2.2.1" targetFramework="net46" />
  <package id="Microsoft.AspNet.Mvc" version="5.2.3" targetFramework="net46" />
  <package id="Microsoft.AspNet.Razor" version="3.2.3" targetFramework="net46" />
  <package id="Microsoft.AspNet.SignalR" version="2.2.2" targetFramework="net46" />
  <package id="Microsoft.AspNet.SignalR.Core" version="2.2.2" targetFramework="net46" />
  <package id="Microsoft.AspNet.SignalR.JS" version="2.2.2" targetFramework="net46" />
  <package id="Microsoft.AspNet.SignalR.SystemWeb" version="2.2.2" targetFramework="net46" />
  <package id="Microsoft.AspNet.Web.Optimization" version="1.1.3" targetFramework="net46" />
  <package id="Microsoft.AspNet.WebPages" version="3.2.3" targetFramework="net46" />
  <package id="Microsoft.Azure.Common" version="2.1.4" targetFramework="net46" />
  <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net46" />
  <package id="Microsoft.Azure.KeyVault" version="2.3.2" targetFramework="net46" />
  <package id="Microsoft.Azure.KeyVault.Core" version="2.0.4" targetFramework="net46" />
  <package id="Microsoft.Azure.KeyVault.WebKey" version="2.0.7" targetFramework="net46" />
  <package id="Microsoft.Bcl" version="1.1.10" targetFramework="net46" />
  <package id="Microsoft.Bcl.Async" version="1.0.168" targetFramework="net46" />
  <package id="Microsoft.Bcl.Build" version="1.0.21" targetFramework="net46" />
  <package id="Microsoft.CodeAnalysis.Analyzers" version="1.1.0" targetFramework="net46" />
  <package id="Microsoft.CodeAnalysis.Common" version="2.3.0" targetFramework="net46" />
  <package id="Microsoft.CodeAnalysis.CSharp" version="2.3.0" targetFramework="net46" />
  <package id="Microsoft.CodeDom.Providers.DotNetCompilerPlatform" version="1.0.5" targetFramework="net46" />
  <package id="Microsoft.Data.Edm" version="5.8.2" targetFramework="net46" />
  <package id="Microsoft.Data.OData" version="5.8.2" targetFramework="net46" />
  <package id="Microsoft.Data.Services.Client" version="5.8.2" targetFramework="net46" />
  <package id="Microsoft.Extensions.DependencyInjection" version="1.1.1" targetFramework="net46" />
  <package id="Microsoft.Extensions.DependencyInjection.Abstractions" version="1.1.1" targetFramework="net46" />
  <package id="Microsoft.IdentityModel.Protocol.Extensions" version="1.0.4.403061554" targetFramework="net46" />
  <package id="Microsoft.jQuery.Unobtrusive.Validation" version="3.2.3" targetFramework="net46" />
  <package id="Microsoft.Net.Compilers" version="2.3.0" targetFramework="net46" developmentDependency="true" />
  <package id="Microsoft.Net.Http" version="2.2.29" targetFramework="net46" />
  <package id="Microsoft.NETCore.Platforms" version="1.1.0" targetFramework="net46" />
  <package id="Microsoft.Orleans.Client" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.Core" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.OrleansAzureUtils" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.OrleansCodeGenerator" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.OrleansProviders" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.OrleansRuntime" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Owin" version="3.1.0" targetFramework="net46" />
  <package id="Microsoft.Owin.Host.SystemWeb" version="3.1.0" targetFramework="net46" />
  <package id="Microsoft.Owin.Security" version="3.1.0" targetFramework="net46" />
  <package id="Microsoft.Owin.Security.ActiveDirectory" version="3.1.0" targetFramework="net46" />
  <package id="Microsoft.Owin.Security.Cookies" version="3.1.0" targetFramework="net46" />
  <package id="Microsoft.Owin.Security.Jwt" version="3.1.0" targetFramework="net46" />
  <package id="Microsoft.Owin.Security.OAuth" version="3.1.0" targetFramework="net46" />
  <package id="Microsoft.Owin.Security.OpenIdConnect" version="3.1.0" targetFramework="net46" />
  <package id="Microsoft.Rest.ClientRuntime" version="2.3.8" targetFramework="net46" />
  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.8" targetFramework="net46" />
  <package id="Microsoft.Web.Infrastructure" version="1.0.0.0" targetFramework="net46" />
  <package id="Microsoft.Win32.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="Microsoft.WindowsAzure.ConfigurationManager" version="3.2.3" targetFramework="net46" />
  <package id="Moment.js" version="2.18.2" targetFramework="net46" />
  <package id="NETStandard.Library" version="1.6.1" targetFramework="net46" />
  <package id="Newtonsoft.Json" version="10.0.3" targetFramework="net46" />
  <package id="Owin" version="1.0" targetFramework="net46" />
  <package id="RazorGenerator.MsBuild" version="2.5.0" targetFramework="net46" developmentDependency="true" />
  <package id="RazorGenerator.Mvc" version="2.4.9" targetFramework="net46" />
  <package id="RecaptchaNet" version="2.1.0" targetFramework="net46" />
  <package id="System.AppContext" version="4.3.0" targetFramework="net46" />
  <package id="System.Collections" version="4.3.0" targetFramework="net46" />
  <package id="System.Collections.Concurrent" version="4.3.0" targetFramework="net46" />
  <package id="System.Collections.Immutable" version="1.3.1" targetFramework="net46" />
  <package id="System.ComponentModel" version="4.3.0" targetFramework="net46" />
  <package id="System.ComponentModel.EventBasedAsync" version="4.3.0" targetFramework="net46" />
  <package id="System.Console" version="4.3.0" targetFramework="net46" />
  <package id="System.Diagnostics.Debug" version="4.3.0" targetFramework="net46" />
  <package id="System.Diagnostics.DiagnosticSource" version="4.4.0" targetFramework="net46" />
  <package id="System.Diagnostics.FileVersionInfo" version="4.3.0" targetFramework="net46" />
  <package id="System.Diagnostics.StackTrace" version="4.3.0" targetFramework="net46" />
  <package id="System.Diagnostics.Tools" version="4.3.0" targetFramework="net46" />
  <package id="System.Diagnostics.Tracing" version="4.3.0" targetFramework="net46" />
  <package id="System.Dynamic.Runtime" version="4.3.0" targetFramework="net46" />
  <package id="System.Globalization" version="4.3.0" targetFramework="net46" />
  <package id="System.Globalization.Calendars" version="4.3.0" targetFramework="net46" />
  <package id="System.IdentityModel.Tokens.Jwt" version="4.0.4.403061554" targetFramework="net46" />
  <package id="System.IO" version="4.3.0" targetFramework="net46" />
  <package id="System.IO.Compression" version="4.3.0" targetFramework="net46" />
  <package id="System.IO.Compression.ZipFile" version="4.3.0" targetFramework="net46" />
  <package id="System.IO.FileSystem" version="4.3.0" targetFramework="net46" />
  <package id="System.IO.FileSystem.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="System.Linq" version="4.3.0" targetFramework="net46" />
  <package id="System.Linq.Expressions" version="4.3.0" targetFramework="net46" />
  <package id="System.Linq.Queryable" version="4.3.0" targetFramework="net46" />
  <package id="System.Net.Http" version="4.3.2" targetFramework="net46" />
  <package id="System.Net.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="System.Net.Requests" version="4.3.0" targetFramework="net46" />
  <package id="System.Net.Sockets" version="4.3.0" targetFramework="net46" />
  <package id="System.ObjectModel" version="4.3.0" targetFramework="net46" />
  <package id="System.Reflection" version="4.3.0" targetFramework="net46" />
  <package id="System.Reflection.Extensions" version="4.3.0" targetFramework="net46" />
  <package id="System.Reflection.Metadata" version="1.4.2" targetFramework="net46" />
  <package id="System.Reflection.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="System.Resources.ResourceManager" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime.Extensions" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime.Handles" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime.InteropServices" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime.InteropServices.RuntimeInformation" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime.Numerics" version="4.3.0" targetFramework="net46" />
  <package id="System.Security.Cryptography.Algorithms" version="4.3.0" targetFramework="net46" requireReinstallation="true" />
  <package id="System.Security.Cryptography.Encoding" version="4.3.0" targetFramework="net46" />
  <package id="System.Security.Cryptography.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="System.Security.Cryptography.X509Certificates" version="4.3.0" targetFramework="net46" requireReinstallation="true" />
  <package id="System.Spatial" version="5.8.2" targetFramework="net46" />
  <package id="System.Text.Encoding" version="4.3.0" targetFramework="net46" />
  <package id="System.Text.Encoding.CodePages" version="4.3.0" targetFramework="net46" />
  <package id="System.Text.Encoding.Extensions" version="4.3.0" targetFramework="net46" />
  <package id="System.Text.RegularExpressions" version="4.3.0" targetFramework="net46" />
  <package id="System.Threading" version="4.3.0" targetFramework="net46" />
  <package id="System.Threading.Tasks" version="4.3.0" targetFramework="net46" />
  <package id="System.Threading.Tasks.Parallel" version="4.3.0" targetFramework="net46" />
  <package id="System.Threading.Thread" version="4.3.0" targetFramework="net46" />
  <package id="System.Threading.Timer" version="4.3.0" targetFramework="net46" />
  <package id="System.ValueTuple" version="4.3.1" targetFramework="net46" />
  <package id="System.Xml.ReaderWriter" version="4.3.0" targetFramework="net46" />
  <package id="System.Xml.XDocument" version="4.3.0" targetFramework="net46" />
  <package id="System.Xml.XmlDocument" version="4.3.0" targetFramework="net46" />
  <package id="System.Xml.XPath" version="4.3.0" targetFramework="net46" />
  <package id="System.Xml.XPath.XDocument" version="4.3.0" targetFramework="net46" />
  <package id="WebActivatorEx" version="2.2.0" targetFramework="net46" />
  <package id="WebGrease" version="1.6.0" targetFramework="net46" />
  <package id="WindowsAzure.Storage" version="8.2.0" targetFramework="net46" />
  <package id="Wivuu.DataSeed" version="1.0.8" targetFramework="net46" />
  <package id="X.PagedList" version="7.1.0" targetFramework="net46" />
  <package id="X.PagedList.Mvc" version="7.1.1" targetFramework="net46" />
</packages>
```

**Comment by onionhammer** [comment]
All packages used by the client which is erroring out

```
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="Antlr" version="3.5.0.2" targetFramework="net461" />
  <package id="Autofac" version="3.5.2" targetFramework="net461" />
  <package id="Autofac.Mvc5" version="3.3.4" targetFramework="net461" />
  <package id="AutoMapper" version="4.1.1" targetFramework="net461" />
  <package id="BuildWebCompiler" version="1.11.347" targetFramework="net461" />
  <package id="Chart.js" version="1.0.2" targetFramework="net461" />
  <package id="EntityFramework" version="6.1.3" targetFramework="net461" />
  <package id="FontAwesome" version="4.7.0" targetFramework="net461" />
  <package id="Hyak.Common" version="1.1.3" targetFramework="net461" />
  <package id="jQuery" version="2.2.4" targetFramework="net461" />
  <package id="jQuery.Validation" version="1.16.0" targetFramework="net461" />
  <package id="LinqKit.EntityFramework" version="1.1.9.0" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights" version="2.3.0" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.Agent.Intercept" version="2.0.7" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.DependencyCollector" version="2.3.0" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.JavaScript" version="0.22.9-build00167" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.PerfCounterCollector" version="2.3.0" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.Web" version="2.3.0" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.Web.TelemetryChannel" version="1.1.0" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.WindowsServer" version="2.3.0" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel" version="2.3.0" targetFramework="net461" />
  <package id="Microsoft.AspNet.Identity.Core" version="2.2.1" targetFramework="net461" />
  <package id="Microsoft.AspNet.Identity.Owin" version="2.2.1" targetFramework="net461" />
  <package id="Microsoft.AspNet.Mvc" version="5.2.3" targetFramework="net461" />
  <package id="Microsoft.AspNet.Razor" version="3.2.3" targetFramework="net461" />
  <package id="Microsoft.AspNet.SignalR" version="2.2.2" targetFramework="net461" />
  <package id="Microsoft.AspNet.SignalR.Core" version="2.2.2" targetFramework="net461" />
  <package id="Microsoft.AspNet.SignalR.JS" version="2.2.2" targetFramework="net461" />
  <package id="Microsoft.AspNet.SignalR.SystemWeb" version="2.2.2" targetFramework="net461" />
  <package id="Microsoft.AspNet.Web.Optimization" version="1.1.3" targetFramework="net461" />
  <package id="Microsoft.AspNet.WebPages" version="3.2.3" targetFramework="net461" />
  <package id="Microsoft.Azure.Common" version="2.1.4" targetFramework="net461" />
  <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net461" />
  <package id="Microsoft.Azure.KeyVault" version="2.3.2" targetFramework="net461" />
  <package id="Microsoft.Azure.KeyVault.Core" version="2.0.4" targetFramework="net461" />
  <package id="Microsoft.Azure.KeyVault.WebKey" version="2.0.7" targetFramework="net461" />
  <package id="Microsoft.Bcl" version="1.1.10" targetFramework="net461" />
  <package id="Microsoft.Bcl.Async" version="1.0.168" targetFramework="net461" />
  <package id="Microsoft.Bcl.Build" version="1.0.21" targetFramework="net461" />
  <package id="Microsoft.CodeAnalysis.Analyzers" version="1.1.0" targetFramework="net461" />
  <package id="Microsoft.CodeAnalysis.Common" version="2.2.0" targetFramework="net461" />
  <package id="Microsoft.CodeAnalysis.CSharp" version="2.2.0" targetFramework="net461" />
  <package id="Microsoft.CodeDom.Providers.DotNetCompilerPlatform" version="1.0.5" targetFramework="net461" />
  <package id="Microsoft.Data.Edm" version="5.8.2" targetFramework="net461" />
  <package id="Microsoft.Data.OData" version="5.8.2" targetFramework="net461" />
  <package id="Microsoft.Data.Services.Client" version="5.8.2" targetFramework="net461" />
  <package id="Microsoft.Extensions.DependencyInjection" version="1.1.1" targetFramework="net461" />
  <package id="Microsoft.Extensions.DependencyInjection.Abstractions" version="1.1.1" targetFramework="net461" />
  <package id="Microsoft.IdentityModel.Protocol.Extensions" version="1.0.4.403061554" targetFramework="net461" />
  <package id="Microsoft.jQuery.Unobtrusive.Validation" version="3.2.3" targetFramework="net461" />
  <package id="Microsoft.Net.Compilers" version="2.2.0" targetFramework="net461" developmentDependency="true" />
  <package id="Microsoft.Net.Http" version="2.2.29" targetFramework="net461" />
  <package id="Microsoft.NETCore.Platforms" version="1.1.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.Client" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.Core" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.OrleansAzureUtils" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.OrleansCodeGenerator" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.OrleansProviders" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Orleans.OrleansRuntime" version="1.5.0" targetFramework="net461" />
  <package id="Microsoft.Owin" version="3.1.0" targetFramework="net461" />
  <package id="Microsoft.Owin.Host.SystemWeb" version="3.1.0" targetFramework="net461" />
  <package id="Microsoft.Owin.Security" version="3.1.0" targetFramework="net461" />
  <package id="Microsoft.Owin.Security.ActiveDirectory" version="3.1.0" targetFramework="net461" />
  <package id="Microsoft.Owin.Security.Cookies" version="3.1.0" targetFramework="net461" />
  <package id="Microsoft.Owin.Security.Jwt" version="3.1.0" targetFramework="net461" />
  <package id="Microsoft.Owin.Security.OAuth" version="3.1.0" targetFramework="net461" />
  <package id="Microsoft.Owin.Security.OpenIdConnect" version="3.1.0" targetFramework="net461" />
  <package id="Microsoft.Rest.ClientRuntime" version="2.3.8" targetFramework="net461" />
  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.8" targetFramework="net461" />
  <package id="Microsoft.Web.Infrastructure" version="1.0.0.0" targetFramework="net461" />
  <package id="Microsoft.Win32.Primitives" version="4.3.0" targetFramework="net461" />
  <package id="Microsoft.WindowsAzure.ConfigurationManager" version="3.2.3" targetFramework="net461" />
  <package id="Moment.js" version="2.18.2" targetFramework="net461" />
  <package id="NETStandard.Library" version="1.6.1" targetFramework="net461" />
  <package id="Newtonsoft.Json" version="10.0.3" targetFramework="net461" />
  <package id="Owin" version="1.0" targetFramework="net461" />
  <package id="RazorGenerator.MsBuild" version="2.5.0" targetFramework="net461" developmentDependency="true" />
  <package id="RazorGenerator.Mvc" version="2.4.9" targetFramework="net461" />
  <package id="RecaptchaNet" version="2.1.0" targetFramework="net461" />
  <package id="System.AppContext" version="4.3.0" targetFramework="net461" />
  <package id="System.Collections" version="4.3.0" targetFramework="net461" />
  <package id="System.Collections.Concurrent" version="4.3.0" targetFramework="net461" />
  <package id="System.Collections.Immutable" version="1.3.1" targetFramework="net461" />
  <package id="System.ComponentModel" version="4.3.0" targetFramework="net461" />
  <package id="System.ComponentModel.EventBasedAsync" version="4.3.0" targetFramework="net461" />
  <package id="System.Console" version="4.3.0" targetFramework="net461" />
  <package id="System.Diagnostics.Debug" version="4.3.0" targetFramework="net461" />
  <package id="System.Diagnostics.DiagnosticSource" version="4.4.0" targetFramework="net461" />
  <package id="System.Diagnostics.FileVersionInfo" version="4.3.0" targetFramework="net461" />
  <package id="System.Diagnostics.StackTrace" version="4.3.0" targetFramework="net461" />
  <package id="System.Diagnostics.Tools" version="4.3.0" targetFramework="net461" />
  <package id="System.Diagnostics.Tracing" version="4.3.0" targetFramework="net461" />
  <package id="System.Dynamic.Runtime" version="4.3.0" targetFramework="net461" />
  <package id="System.Globalization" version="4.3.0" targetFramework="net461" />
  <package id="System.Globalization.Calendars" version="4.3.0" targetFramework="net461" />
  <package id="System.IdentityModel.Tokens.Jwt" version="4.0.4.403061554" targetFramework="net461" />
  <package id="System.IO" version="4.3.0" targetFramework="net461" />
  <package id="System.IO.Compression" version="4.3.0" targetFramework="net461" />
  <package id="System.IO.Compression.ZipFile" version="4.3.0" targetFramework="net461" />
  <package id="System.IO.FileSystem" version="4.3.0" targetFramework="net461" />
  <package id="System.IO.FileSystem.Primitives" version="4.3.0" targetFramework="net461" />
  <package id="System.Linq" version="4.3.0" targetFramework="net461" />
  <package id="System.Linq.Expressions" version="4.3.0" targetFramework="net461" />
  <package id="System.Linq.Queryable" version="4.3.0" targetFramework="net461" />
  <package id="System.Net.Http" version="4.3.2" targetFramework="net461" />
  <package id="System.Net.Primitives" version="4.3.0" targetFramework="net461" />
  <package id="System.Net.Requests" version="4.3.0" targetFramework="net461" />
  <package id="System.Net.Sockets" version="4.3.0" targetFramework="net461" />
  <package id="System.ObjectModel" version="4.3.0" targetFramework="net461" />
  <package id="System.Reflection" version="4.3.0" targetFramework="net461" />
  <package id="System.Reflection.Extensions" version="4.3.0" targetFramework="net461" />
  <package id="System.Reflection.Metadata" version="1.4.2" targetFramework="net461" />
  <package id="System.Reflection.Primitives" version="4.3.0" targetFramework="net461" />
  <package id="System.Resources.ResourceManager" version="4.3.0" targetFramework="net461" />
  <package id="System.Runtime" version="4.3.0" targetFramework="net461" />
  <package id="System.Runtime.Extensions" version="4.3.0" targetFramework="net461" />
  <package id="System.Runtime.Handles" version="4.3.0" targetFramework="net461" />
  <package id="System.Runtime.InteropServices" version="4.3.0" targetFramework="net461" />
  <package id="System.Runtime.InteropServices.RuntimeInformation" version="4.3.0" targetFramework="net461" />
  <package id="System.Runtime.Numerics" version="4.3.0" targetFramework="net461" />
  <package id="System.Security.Cryptography.Algorithms" version="4.3.0" targetFramework="net461" />
  <package id="System.Security.Cryptography.Encoding" version="4.3.0" targetFramework="net461" />
  <package id="System.Security.Cryptography.Primitives" version="4.3.0" targetFramework="net461" />
  <package id="System.Security.Cryptography.X509Certificates" version="4.3.0" targetFramework="net461" />
  <package id="System.Spatial" version="5.8.2" targetFramework="net461" />
  <package id="System.Text.Encoding" version="4.3.0" targetFramework="net461" />
  <package id="System.Text.Encoding.CodePages" version="4.3.0" targetFramework="net461" />
  <package id="System.Text.Encoding.Extensions" version="4.3.0" targetFramework="net461" />
  <package id="System.Text.RegularExpressions" version="4.3.0" targetFramework="net461" />
  <package id="System.Threading" version="4.3.0" targetFramework="net461" />
  <package id="System.Threading.Tasks" version="4.3.0" targetFramework="net461" />
  <package id="System.Threading.Tasks.Parallel" version="4.3.0" targetFramework="net461" />
  <package id="System.Threading.Thread" version="4.3.0" targetFramework="net461" />
  <package id="System.Threading.Timer" version="4.3.0" targetFramework="net461" />
  <package id="System.ValueTuple" version="4.3.1" targetFramework="net461" />
  <package id="System.Xml.ReaderWriter" version="4.3.0" targetFramework="net461" />
  <package id="System.Xml.XDocument" version="4.3.0" targetFramework="net461" />
  <package id="System.Xml.XmlDocument" version="4.3.0" targetFramework="net461" />
  <package id="System.Xml.XPath" version="4.3.0" targetFramework="net461" />
  <package id="System.Xml.XPath.XDocument" version="4.3.0" targetFramework="net461" />
  <package id="WebActivatorEx" version="2.2.0" targetFramework="net461" />
  <package id="WebGrease" version="1.6.0" targetFramework="net461" />
  <package id="WindowsAzure.Storage" version="8.2.0" targetFramework="net461" />
  <package id="Wivuu.DataSeed" version="1.0.8" targetFramework="net461" />
  <package id="X.PagedList" version="7.1.0" targetFramework="net461" />
  <package id="X.PagedList.Mvc" version="7.1.1" targetFramework="net461" />
</packages>
```

**Comment by onionhammer** [comment]
Hey guys, I got this working when I upgraded the `Microsoft.Net.Compilers` from 2.2.0 to 2.3.1. After that I got a different error ("Must include private members unless emitting a ref assembly"), but adding `Microsoft.Orleans.OrleansCodeGenerator.Build` to the client project solved that.

**Comment by galvesribeiro** [comment]
Humm... @reubenbond any idea?

**Comment by ReubenBond** [comment]
I have no idea. It's the kind of error we might see if for some reason the wrong mscorlib was being referenced - because of all the new .NET magic with 'ref assemblies', but I don't know how to reason about it.

I wonder if this happens if you upgrade from a previous version of .NET to 4.6.x without performing a reinstall of all NuGet packages?

**Comment by onionhammer** [comment]
No, I reinstalled all the nuget packages

**Comment by ReubenBond** [comment]
@onionhammer the [aforementioned issue](https://github.com/dotnet/orleans/issues/3217#issuecomment-318711837) will be fixed in 1.5.1, so I believe we can close this. Reopen it if you feel it's not resolved.

**Comment by onionhammer** [comment]
@ReubenBond thanks!

**Comment by onionhammer** [comment]
@ReubenBond This (or something very similar) is still an issue. If we don't have `Microsoft.Orleans.OrleansCodeGenerator.Build` referenced, we see 

![image](https://user-images.githubusercontent.com/969938/30455208-a51ea7d0-9964-11e7-951d-7d0f36de3aa7.png)

**Comment by sergeybykov** [comment]
We plan to remove run time codegen in 2.0, so referencing `Microsoft.Orleans.OrleansCodeGenerator.Build` will be required. @onionhammer will that solve the issue for you?

**Comment by sergeybykov** [comment]
Closing because we released 2.0.0-beta with reworked codegen.
